climakitae/new_core/processors/warming_level.py

```diff
@@ -0,0 +1,424 @@
+    organized by warming levels, following the established warming level methodology.
+
+    Parameters
+    ----------
+    value : Dict[str, Any]
+        Configuration dictionary containing:
+        - warming_levels : list[float]
+            List of global warming levels in degrees C (e.g., [1.5, 2.0])
+        - warming_level_months : list[int], optional
+            List of months to include (1-12). Default: all months
+        - warming_level_window : int, optional
+            Number of years before and after the central year. Default: 15
+
+    Methods
+    -------
+    execute : Transform data to warming level approach
+    update_context : Update processing context with warming level information
+    set_data_accessor : Set data catalog accessor
+
+    Notes
+    -----
+    The input data must span from 1980-2100 and include historical climate data
+    for proper warming level calculations. Data should have simulation and scenario
+    dimensions or be properly configured for stacking.
+    """
+
+    def __init__(self, value: Dict[str, Any]):
+        """
+        Initialize the warming level processor.
+
+        Parameters
+        ----------
+        value : Dict[str, Any]
+            Configuration dictionary with warming level parameters.
+        """
+        # Validate input
+        if not isinstance(value, dict):
+            raise TypeError("Expected dictionary for warming level configuration")
+
+        # Extract configuration parameters
+        self.warming_levels = value.get("warming_levels", [2.0])
+        self.warming_level_window = value.get("warming_level_window", 15)
+
+        # Initialize instance variables
+        self.name = "warming_level_simple"
+        self.warming_level_times = read_csv_file(
+            GWL_1850_1900_FILE, index_col=[0, 1, 2], parse_dates=True
+        )
+        self.warming_level_times_idx = read_csv_file(
+            GWL_1981_2010_TIMEIDX_FILE, index_col="time", parse_dates=True
+        )
+        self.catalog = None
+        self.value = {
+            "warming_levels": self.warming_levels,
+            "warming_level_window": self.warming_level_window,
+        }
+
+    def execute(
+        self,
+        result: Union[
+            xr.Dataset, xr.DataArray, Iterable[Union[xr.Dataset, xr.DataArray]]
+        ],
+        context: Dict[str, Any],
+    ) -> Union[xr.Dataset, xr.DataArray, Iterable[Union[xr.Dataset, xr.DataArray]]]:
+        """
+        Transform time-series data to warming level approach.
+
+        1. find (from precomputed values) when a given warming level is reached by a
+            simulation (GCM, run, scenario)
+            a. some fancy handling of values between precomputed values happens?
+        2. slice in a window of X years around that date
+            a. if the slice has a start date earlier than the simulation data, splice
+                historical onto the slice for the requested variable
+        3. return the data
```

Comment:
Sorry Calvin, I think this is my old explanation of what the order of operations is. Can you clean this up?